kafka/conn.py

```diff
@@ -120,11 +120,12 @@ def connect(self):
         if self.state is ConnectionStates.CONNECTING:
             # in non-blocking mode, use repeated calls to socket.connect_ex
             # to check connection status
+            ret = None
             request_timeout = self.config['request_timeout_ms'] / 1000.0
             try:
```

Comment:
I dont think we even need the try / except here - connect_ex should never raise an exception.